File: Backend/src/main/java/backend/osiris/entity/inventory/Transfer.java
package backend.osiris.entity.inventory;

import backend.osiris.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Accessors(chain = true)
@Entity
@Table(name = "transfer")
public class Transfer extends BaseEntity {
    @Column(name = "code", nullable = false, unique = true)
    private String code;

    @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JoinColumn(name = "export_docket_id", referencedColumnName = "id", nullable = false, unique = true)
    private Docket exportDocket;

    @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JoinColumn(name = "import_docket_id", referencedColumnName = "id", nullable = false, unique = true)
    private Docket importDocket;

    @Column(name = "note")
    private String note;
}

File: Frontend/src/models/Transfer.ts
import BaseResponse from "./BaseResponse";
import { DocketResponse, DocketRequest } from "./Docket";

export interface TransferResponse extends BaseResponse {
    code: string;
    exportDocket: DocketResponse;
    importDocket: DocketResponse;
    note: string | null;
}

export interface TransferRequest {
    code: string;
    exportDocket: DocketRequest;
    importDocket: DocketRequest;
    note: string | null;
}
